### ISSUE TITLE
List of function properties should put the Function.prototype methods at the bottom of the list

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#20283]
_From @ORESoftware on November 27, 2017 3:4_

This is a feature request - the problem is demonstrated in this video

basically, we have  a function

```js
interface ItHook {
  (t) => any;
  foo: 3;
  bar: true
}
```

when we list the function properties, it will list the `Function.prototype` methods before it lists foo or bar, etc.

https://www.useloom.com/share/1a0f1e84083049ff8f745a3a1f17ac03

You can go to 2:53 in the video to see the problem.

https://github.com/Microsoft/TypeScript/issues/20274

_Copied from original issue: Microsoft/vscode#39160_

**Comment by mjbvz** [comment]
For the code:

```ts
interface ItHook {
  (t): any;
  foo: 3;
  bar: true
}

function foo(x: ItHook) {
  x.
}
```

All entries currently have the same sortText:

```
[Trace  - 13:56:37] Response received: completions (206). Request took 2 ms. Success: true 
Result: [
    {
        "name": "apply",
        "kind": "method",
        "kindModifiers": "declare",
        "sortText": "0"
    },
    {
        "name": "arguments",
        "kind": "property",
        "kindModifiers": "declare",
        "sortText": "0"
    },
    {
        "name": "bar",
        "kind": "property",
        "kindModifiers": "",
        "sortText": "0"
    },
    {
        "name": "bind",
        "kind": "method",
        "kindModifiers": "declare",
        "sortText": "0"
    },
    {
        "name": "call",
        "kind": "method",
        "kindModifiers": "declare",
        "sortText": "0"
    },
    {
        "name": "caller",
        "kind": "property",
        "kindModifiers": "declare",
        "sortText": "0"
    },
    {
        "name": "foo",
        "kind": "property",
        "kindModifiers": "",
        "sortText": "0"
    },
    {
        "name": "length",
        "kind": "property",
        "kindModifiers": "declare",
        "sortText": "0"
    },
    {
        "name": "name",
        "kind": "property",
        "kindModifiers": "declare",
        "sortText": "0"
    },
    {
        "name": "prototype",
        "kind": "property",
        "kindModifiers": "declare",
        "sortText": "0"
    },
    {
        "name": "toString",
        "kind": "method",
        "kindModifiers": "declare",
        "sortText": "0"
    }
]
```

So we end up sorting alphabetically. It may make sense to show sort prototype properties lower in this case

**Comment by ORESoftware** [comment]
yes I think it would help to show them last.
perhaps show the type - `typeof === 'function'` at the top, but show the prototype properties at the bottom.

**Comment by RyanCavanaugh** [comment]
@mhegazy I'd say :shipit: - thoughts?

**Comment by mhegazy** [comment]
We have not changed the sort order historically.. we only omitted members from the list, e.g. built-in `Object` members. we did not do that for `Function`, but that was more of a bug than a feature, and ppl seemed to like it and did not mind the additional entries. i personally find changing the order a bit disorienting, and i always assume the list is ordered alphabetically. I would like to know if other users found that the additional entries are too much noise given that we hear ppl complaining about  missing properties from on `Object`.
